src/serial.rs

```diff
@@ -320,12 +320,9 @@ trait UsartReadWrite: Deref<Target = crate::pac::usart1::RegisterBlock> {
 
         if let Some(err) = err {
             // Some error occurred. In order to clear that error flag, you have to
-            // do a read from the sr register followed by a read from the dr
-            // register
-            unsafe {
-                ptr::read_volatile(&self.sr as *const _ as *const _);
-                self.dr.read().dr().bits();
-            }
+            // do a read from the sr register (already done above) followed by 
+            // a read from the dr register
+            self.dr.read();
             Err(nb::Error::Other(err))
```

Comment:
Can you explain why you've deleted this?

Reply:
> Can you explain why you've deleted this?

I did not find in the documentation for the microcontroller that reading from the SR register should be performed twice.

Reply:
cc @TheZoq2 
https://github.com/japaric/stm32f103xx-hal/commit/fae894cccae386c0b1650f32eb038e0089b68279

Reply:
I restored the reading from the SR register.